flake.nix

```diff
@@ -0,0 +1,149 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=ea5234e7073d5f44728c499192544a84244bf35a";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" "x86_64-windows" "x86_64-darwin" "aarch64-darwin" ] (targetSystem:
+      let
+        # This needs to be made dynamic using something like builtins.currentSystem
+        buildSystem = "x86_64-linux";
+
+        systems = {
+          "x86_64-linux" = [ "linux" "x64" ];
+          "x86_64-windows" = [ "win32" "x64" ];
+          "x86_64-darwin" = [ "darwin" "x64" ];
+          "aarch64-darwin" = [ "darwin" "arm64" ];
+        };
```

Comment:
For now it does not make sense to have separate build systems, we are not doing any kind of cross compilation, can you simplify this for now.

Cross compilation is entirely separate problem and should not be solved here.

PK CLI is currently a JS codebase, which means it is expected to run on Nodejs. Cross compilation of node is an entirely separate problem to be put into another project.

Reply:
What I have done here is replicating the original `release.nix`. Before you could choose what platform to build a binary for, and in this case I implemented it as such; the only requirement being running the build process on a `x86_64-linux` system. There is no "cross-compilation" when it comes to making builds on any device, its strictly limited to Linux devices, but those devices are capable of leveraging `vercel/pkg` to build for other specific platforms.

But if this is not desired right now, then it is fine for it to be simplified down. Will get this done.

Reply:
> Right so in the case of PK CLI, the only architecture we are using to "build" PK CLI is the x86_64-linux.

Add commentary about how to expand this if this changes in the future.

Reply:
Okay, as of commit 1d7af25 this should be made a lot more clear through the code refactoring and comments. I have not removed the functionality to build for other systems, and it is still forced to use the `x86_64-linux` package set for `nixpkgs` when building. So this should be identical in functionality to the original build system in staging now.